google/auth/transport/aiohttp_req.py

```diff
@@ -0,0 +1,414 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transport adapter for Async HTTP (aiohttp)."""
+
+from __future__ import absolute_import
+
+import asyncio
+import functools
+import logging
+import numbers
+import time
+
+import aiohttp
+import six
+
+# import google.auth
+from google.auth import exceptions
+from google.auth import transport
+
+# import google.auth.transport._mtls_helper
+
+
+_OAUTH_SCOPES = [
+    "https://www.googleapis.com/auth/appengine.apis",
+    "https://www.googleapis.com/auth/userinfo.email",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEFAULT_TIMEOUT = 120  # in seconds
```

Comment:
is this the same timeout as async? if so we should avoid copying instead sharing timeouts.

Reply:
The asyncio timeout exception is different, so we're forced to write a new Timeout Guard to deal with that